src/check/parse/tokenize/src/main.zig

```diff
@@ -0,0 +1,1399 @@
+    }
+
+    pub fn chompIntegerBase10(self: *Cursor) void {
+        while (self.peek()) |c| {
+            if (c >= '0' and c <= '9') {
+                self.pos += 1;
+            } else if (c == '_') {
+                self.pos += 1;
+            } else {
+                break;
+            }
+        }
+    }
+
+    pub fn chompIntegerBase16(self: *Cursor) void {
+        while (self.peek()) |c| {
+            if ((c >= '0' and c <= '9') or (c >= 'a' and c <= 'f') or (c >= 'A' and c <= 'F')) {
+                self.pos += 1;
+            } else if (c == '_') {
+                self.pos += 1;
+            } else {
+                break;
+            }
+        }
+    }
+
+    pub fn chompIntegerBase8(self: *Cursor) void {
+        while (self.peek()) |c| {
+            if (c >= '0' and c <= '7') {
+                self.pos += 1;
+            } else if (c == '_') {
+                self.pos += 1;
+            } else {
+                break;
+            }
+        }
+    }
+
+    pub fn chompIntegerBase2(self: *Cursor) void {
+        while (self.peek()) |c| {
+            if (c == '0' or c == '1') {
+                self.pos += 1;
+            } else if (c == '_') {
+                self.pos += 1;
+            } else {
+                break;
+            }
+        }
+    }
+
+    pub fn chompIdentLower(self: *Cursor) T {
+        const start = self.pos;
+        var kwCheck: bool = true;
+        while (self.peek()) |c| {
+            if (c >= 'a' and c <= 'z') {
+                self.pos += 1;
+            } else if ((c >= 'A' and c <= 'Z') or (c >= '0' and c <= '9')) {
+                self.pos += 1;
+                kwCheck = false;
+            } else {
+                break;
+            }
+        }
+        if (kwCheck and (self.pos - start) <= 10) {
+            const ident = self.buf[start..self.pos];
+            const kw = Token.getKeyword(ident);
+            return kw orelse T.LowerIdent;
+        } else {
+            return T.LowerIdent;
+        }
+    }
+
+    pub fn chompIdentUpper(self: *Cursor) T {
+        while (self.pos < self.buf.len) {
+            const c = self.buf[self.pos];
+            if ((c >= 'a' and c <= 'z') or (c >= 'A' and c <= 'Z') or (c >= '0' and c <= '9')) {
+                self.pos += 1;
+            } else {
+                break;
+            }
+        }
+        return T.UpperIdent;
+    }
+
+    pub fn chompInteger(self: *Cursor) void {
+        while (self.pos < self.buf.len) {
+            const c = self.buf[self.pos];
+            if (c >= '0' and c <= '9') {
+                self.pos += 1;
+            } else {
+                break;
+            }
+        }
+    }
+
+    pub fn chompIdentGeneral(self: *Cursor) void {
+        while (self.pos < self.buf.len) {
+            const c = self.buf[self.pos];
+            if ((c >= 'a' and c <= 'z') or (c >= 'A' and c <= 'Z') or (c >= '0' and c <= '9') or c == '_') {
+                self.pos += 1;
+            } else {
```

Comment:
Given how many exceptions there are that don't use peek, I really wonder if it would be better to just not have peek at all. The inconsistency makes it feel like it may be hiding bugs in various places.

Reply:
I'd rather see consistent use of peek and advance(to ensure we stay in bounds)